homeassistant/components/vodafone_station/coordinator.py

```diff
@@ -0,0 +1,170 @@
+
+        if dev_info.connected:
+            self._last_activity = utc_point_in_time
+
+        self._connection_type = dev_info.connection_type
+        self._ip_address = dev_info.ip_address
+        self._wifi = dev_info.wifi
+
+    @property
+    def connection_type(self) -> str:
+        """Return connected status."""
+        return self._connection_type or ""
+
+    @property
+    def is_connected(self) -> bool:
+        """Return connected status."""
+        return self._connected
+
+    @property
+    def mac_address(self) -> str:
+        """Get MAC address."""
+        return self._mac
+
+    @property
+    def hostname(self) -> str | None:
+        """Get Name."""
+        return self._name
+
+    @property
+    def ip_address(self) -> str | None:
+        """Get IP address."""
+        return self._ip_address
+
+    @property
+    def last_activity(self) -> datetime | None:
+        """Return device last activity."""
+        return self._last_activity
+
+    @property
+    def wifi(self) -> str | None:
+        """Return device WIFi connection."""
+        return self._wifi
+
+
+class UpdateCoordinatorDataType(TypedDict):
+    """Update coordinator data type."""
+
+    devices: dict[str, VodafoneStationDeviceInfo]
+    sensors: dict[str, Any]
+
+
+class VodafoneStationRouter(DataUpdateCoordinator):
+    """Queries router running Vodafone Station firmware."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        host: str,
+        username: str,
+        password: str,
+        config_entry_unique_id: str | None,
+    ) -> None:
+        """Initialize the scanner."""
+
+        self._host = host
+        self.api = VodafoneStationApi(host, username, password)
+
+        # Last resort as no MAC or S/N can be retrieved via API
+        self._id = config_entry_unique_id
+
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=f"{DOMAIN}-{host}-coordinator",
+            update_interval=timedelta(seconds=30),
+        )
+
+    async def _async_update_data(self) -> UpdateCoordinatorDataType:
+        """Update router data."""
+        _LOGGER.debug("Polling Vodafone Station host: %s", self._host)
+        try:
+            logged = await self.api.login()
+        except (asyncio.exceptions.TimeoutError, aiohttp.ClientConnectorError) as err:
+            _LOGGER.warning("Connection error for %s", self._host)
+            raise UpdateFailed(f"Error fetching data: {repr(err)}") from err
+
+        if not logged:
+            raise ConfigEntryAuthFailed
+
+        data: UpdateCoordinatorDataType = {
+            "devices": {},
+            "sensors": {},
+        }
+        list_devices = await self.api.get_all_devices()
+        dev_info: VodafoneStationDevice
+        for _, dev_info in list_devices.items():
+            dev = VodafoneStationDeviceInfo(dev_info.mac, dev_info.name)
+            dev.update(dev_info)
```

Comment:
I think it would make more sense to pass the dev_info to the constructor instead of passing it in pieces and than updating it

Reply:
Done